upup/pkg/fi/cloudup/auth.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudup
+
+import (
+	api "k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+type AuthTaskBuilder struct {
+}
+
+var _ TaskBuilder = &AuthTaskBuilder{}
+
+var _ loader.OptionsBuilder = &AuthTaskBuilder{}
+
+func (b *AuthTaskBuilder) BuildTasks(l *Loader) error {
+	return nil
+}
+
+func (b *AuthTaskBuilder) BuildOptions(options interface{}) error {
+	cluster := options.(*api.ClusterSpec)
+
+	if cluster.KubeAPIServer == nil {
+		cluster.KubeAPIServer = &api.KubeAPIServerConfig{}
+	}
+
+	if cluster.KubeAPIServer.RuntimeConfig == nil {
+		cluster.KubeAPIServer.RuntimeConfig = make(map[string]string)
+	}
+
+	cluster.KubeAPIServer.RuntimeConfig["rbac.authorization.k8s.io/v1alpha1"] = "true"
+
+	cluster.KubeAPIServer.AuthorizationMode = fi.String("RBAC")
```

Comment:
Not sure exactly where this code is exercised in the flow of `kops edit cluster`, but if this purely sets auth mode to RBAC, aren't system components going to have a hard time working while their service account tokens are rejected?  Won't there be a chicken-egg problem where the operator can't apply the RBAC roles in your examples above to enable system accounts, since the system isn't functioning, since the system accounts haven't been authorized for things yet?